src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -118,19 +119,19 @@ private static bool ReflectionMethodEnabled
             }
         }
 
-        private TempAssembly _tempAssembly;
+        private TempAssembly? _tempAssembly;
 #pragma warning disable 0414
         private bool _typedSerializer;
 #pragma warning restore 0414
-        private readonly Type _primitiveType;
-        private XmlMapping _mapping;
+        private readonly Type? _primitiveType;
+        private XmlMapping _mapping = null!;
         private XmlDeserializationEvents _events;
-        internal string DefaultNamespace;
-        private Type _rootType;
+        internal string? DefaultNamespace;
+        private Type _rootType = null!;
```

Comment:
In line 588, _rootType is being set to `null` (because `type` is `null`). I think `_rootType` should be nullable here.

Reply:
No, I've actually spent a bit of time on this case when annotating. Type cannot be null anywhere here - it will pop up as some sort of error (I think ArgumentNull) if you provide null.

Reply:
Added a comment here: https://github.com/dotnet/runtime/pull/41261/files/ca7e8886321ecf2715be375f7ac9b227580fcb80#r477576596

Reply:
changing to nullable